tests/bls_tests.rs

```diff
@@ -0,0 +1,341 @@
+
+        let signature_bytes = decode_hex_with_prefix(&self.output);
+        let expected_signature = Signature::from_bytes(&signature_bytes).unwrap();
+
+        signature == expected_signature
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+
+struct AggregatingTestIO {
+    input: Vec<String>,
+    output: String,
+}
+
+impl AggregatingTestIO {
+    fn verify_aggregate(&self) -> bool {
+        let input_signatures: Vec<Signature> = self
+            .input
+            .iter()
+            .map(|x| Signature::from_bytes(&decode_hex_with_prefix(&x)).unwrap())
+            .collect();
+        let aggregate = match aggregate(&input_signatures) {
+            Ok(agg) => agg,
+            // handling for zero sized input and output
+            Err(_) => {
+                if self.output == String::from("~") {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        };
+        let expected_aggregate_raw = decode_hex_with_prefix(&self.output);
+        let expected_aggregate = Signature::from_bytes(&expected_aggregate_raw).unwrap();
+        aggregate == expected_aggregate
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+
+struct AggVerifyInput {
+    pubkeys: Vec<String>,
+    messages: Vec<String>,
+    signature: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+
+struct AggVerifyTestIO {
+    input: AggVerifyInput,
+    output: bool,
+}
+
+impl AggVerifyTestIO {
+    fn aggregate_verify(&self) -> bool {
+        let pubkeys_result: Result<Vec<PublicKey>, _> = self
+            .input
+            .pubkeys
+            .iter()
+            .map(|x| PublicKey::from_bytes(&decode_hex_with_prefix(&x)))
+            .collect();
+        let pubkeys = match pubkeys_result {
+            Ok(pk) => pk,
+            // error handling for infinity pub key
+            Err(_) => {
+                if self.output == false {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        };
+        let messages_vec: Vec<_> = self
+            .input
+            .messages
+            .iter()
+            .map(|x| decode_hex_with_prefix(x))
+            .collect();
+        let messages_slice: Vec<&[u8]> = messages_vec.iter().map(|x| x.as_slice()).collect();
+        let signature: Signature =
+            match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
+                Ok(sign) => sign,
+                Err(_) => {
+                    // handling for the zero signature case which raises a blst bad encoding error
+                    if self.output == false {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        let verify_result = aggregate_verify(&pubkeys, messages_slice.as_ref(), signature);
```

Comment:
i'd say this is slightly more preferred in rust land
```suggestion
        let verify_result = aggregate_verify(&pubkeys, &messages_slice, signature);
```